README.md

```diff
@@ -6,6 +6,42 @@ hperf is a tool for active measurements of the maximum achievable bandwidth betw
 Hperf was made to test networks in large infrastructure. It's highly scalable and cabaple of running parallel tests over
 a long period of time. 
 
+## Common use cases
+- Debugging link/nic MTU issues
+- Optimizing throughput speed for specific payload/buffer sizes
+- Finding servers that present latency on the application level when ping is showing no latency
+- Testing overall network throughput
+- Testing server to server connectivity 
+
+## Core concepts
+### The hperf binary
+The binary can act as both client and server.
+
+### Client
+The client part of hperf is responsible for orchestrating the servers. Its only job is to send commands to the
+servers and receive incremental stats updates. It can be executed from any machine that can talk to the servers.
+
+### Servers
+Servers are the machines we are testing. To launch the hperf command in servers mode, simply use the `server` command:
+NOTE: `server` is the only command you can execute on the servers. All other commands are executed from the client.
+```bash
+$ ./hperf server --help
+```
+This command will start an API and websocket on the given `--address` and save test results to `--storage-path`. 
+
+### The listen command
+Hperf has been modified to work very differently than how it was before. The new version can run tests without a
+specific `client` needing to be constantly connected. Once the `client` has started a test, the `client` can 
+easily exit without interrupting the test stopping.
+
+Any `client` can hook into the list test at runtime using the `--id` of the test. There can even be multiple `clients`
+listening to the same test.
+
+Example:
+```bash
+$ ./hperf listen --hosts 10.10.1.{2...10} --id [TEST_ID]
```

Comment:
Maybe post more, so I can see how ID is printed and how to reconnect after ctrl-c.